bin/node/rpc-client/src/main.rs

```diff
@@ -22,21 +22,21 @@
 //! This module shows how you can write a Rust RPC client that connects to a running
 //! substrate node and use statically typed RPC wrappers.
 
-use futures::{Future, TryFutureExt};
-use jsonrpc_core_client::{transports::http, RpcError};
+use futures::TryFutureExt;
+use jsonrpsee::{types::Error, ws_client::WsClientBuilder};
 use node_primitives::Hash;
-use sc_rpc::author::{hash::ExtrinsicOrHash, AuthorClient};
+use sc_rpc::author::{hash::ExtrinsicOrHash, AuthorApiClient};
 
-fn main() -> Result<(), RpcError> {
+#[tokio::main]
+async fn main() -> Result<(), Error> {
 	sp_tracing::try_init_simple();
 
-	futures::executor::block_on(async {
-		let uri = "http://localhost:9933";
-
-		http::connect(uri)
-			.and_then(|client: AuthorClient<Hash, Hash>| remove_all_extrinsics(client))
-			.await
-	})
+	// NOTE(niklasad1): changed this to the WS client because the jsonrpsee proc macros
```

Comment:
```suggestion
	// TODO(niklasad1):  https://github.com/paritytech/jsonrpsee/issues/448 
	// changed this to the WS client because the jsonrpsee proc macros
```